pvlib/tests/test_losses.py

```diff
@@ -90,3 +97,101 @@ def test_soiling_hsu(rainfall_input, expected_output_2):
                          rain_accum_period=pd.Timedelta('3h'))
 
     assert_series_equal(result, expected)
+
+
+@pytest.fixture
+def greensboro_rain():
+    # get TMY3 data with rain
+    greensboro = read_tmy3(DATA_DIR / '723170TYA.CSV', coerce_year=1990)
+    # NOTE: can't use Sand Point, AK b/c Lprecipdepth is -9900, ie: missing
+    return greensboro[0].Lprecipdepth
+
+
+@pytest.fixture
+def expected_kimber_soiling_nowash():
+    return pd.read_csv(
+        DATA_DIR / 'greensboro_kimber_soil_nowash.dat',
+        parse_dates=True, index_col='timestamp')
+
+
+@needs_pandas_0_22
```

Comment:
I didn't get to try this but I assume these `@needs_pandas_0_22` are necessary. If so I'm even more inclined to bump pandas version to unify the output between versions. No change necessary, just thinking out loud mostly.

Reply:
Pandas `DataFrame.rolling()` kwarg `closed` isn't available for [py35-min on Travis](https://travis-ci.org/pvlib/pvlib-python/jobs/650693240#L857) but everything else passes, py35-min uses pandas-0.18.1

Reply:
My understanding is that the function does not work at all with pandas 0.18. Correct? If so, that's a different kind of breakage than we've traditionally allowed across dependency versions. The function documentation doesn't give any hint as to why it's going to fail in that scenario either.

I thought we were in agreement that we'd bump the pandas spec for 0.8, not 0.7.2. This suggests that we need to do it for 0.7.2. Not the end of the world, but not ideal.

Reply:
Oops, I didn't realize that. The existing `soiling_hsu` also used this `closed` kwarg in `DataFrame.rolling()`

https://github.com/pvlib/pvlib-python/blob/64a6498b689d27d77bb5967e13eab9bc0018d68e/pvlib/losses.py#L69

but it's pandas dependency was masked in the test by `required_scipy`

https://github.com/pvlib/pvlib-python/blob/64a6498b689d27d77bb5967e13eab9bc0018d68e/pvlib/tests/test_losses.py#L83

One idea is just dropping this `closed` kwarg from `DataFrame.rolling()` which works fine in pandas-0.18.1, the default behavior is 'both' which isn't the end of the world, but probably not strictly faithful to the references, which imply exclusive of starting timestamp.

Any other ideas @cwhanse? @kanderso_nrel?

Reply:
>These @needs* decorators have traditionally been used for only optional dependencies or quirks between versions (like pandas sum of nans = nan or = 0).

Thanks for clarifying this further for us @wholmgren. 

> I thought we were in agreement that we'd bump the pandas spec for 0.8, not 0.7.2. This suggests that we need to do it for 0.7.2. Not the end of the world, but not ideal.

We can revert this merge and chamber it for v0.8. It'll give time to address some of the issues raised brought up and stay consistent with our plan to bump pandas at v0.8.  What do you think @wholmgren?

Reply:
I don't think we need to revert. @mikofski pointed out that the same issue exists in the already released `soiling_hsu`. We could add a note to the doc string, or a try/except with a better error message around the rolling call, or do nothing. I'm ok with any of those.

We should just be more careful with the requires and needs decorators. I know I've been lazy about them in the past.

This is all also an argument for 1) adding scipy to the base requirements so the test suite doesn't have gotchas like with the hsu function (issue #483) and 2) being more aggressive with bumping dependency specs so that contributors don't have to spend as much time supporting older libraries (issue #828).